### DISCUSSION TITLE
Long assembly and No such file or directory

### DISCUSSION THREAD

**Discussion by asoteamgen** [codemagic-ci-cd/codemagic-docs#2821] [Q&A]
### If you are looking for help, please confirm the following...

- [x] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Unity

### Steps to reproduce

installing unity version

### Expected results

long assembly

### Actual results

using a script to build a unity project for ios which is specified in the documentation, I received an error when activating the license(No such file or directory), it helped to install the unity version and update unity_home, but this significantly increases the build time, how can I avoid installing unity? check my Build id

### Build id (optional)

672ceb412ab34780d53f2cbe

**Comment by dam-ease** [comment]
Hello @asoteamgen,
When you add the unity version to your codemagic.yaml file without running the install step, do you run into any issues at all, if you do can you share your buildID? I think adding any of the unity versions listed [here](https://docs.codemagic.io/knowledge-others/install-unity-version/#quick-install-macos-only) since you are running on macOS machine should work fine.

**Reply by asoteamgen** [reply]
@dam-ease When I add the unity setting to the script and update the UNITY_HOME variable, everything works well, but it takes a lot of time. However, this step is missing in the documentation and I found the information on another documentation page. Without installing a specific version of Unity, I get an error when activating No such file or directory. Without installing Unity via a script, I generally don't see that unity is on the M2 Mac.

672ceb412ab34780d53f2cbe

**Reply by dam-ease** [reply]
Thanks for your response @asoteamgen!

> When I add the unity setting to the script and update the UNITY_HOME variable, everything works well, but it takes a lot of time.

This is a much needed step that can't be skipped due to Unity requirements to run the Unity activate license and other build commands. So it can't be avoided and also the reason you run into the error `No such file or directory` when it's not specified.

I noticed in one of your builds that you were trying to install a version of Unity that wasn't available (2022.3.51f1), if you'd still like to install this, please do let us know and it will be added.

> However, this step is missing in the documentation and I found the information on another documentation page.

Also, can you share more details on which step is missing here, the information to install this is available in the docs I shared above, sharing it again here - https://docs.codemagic.io/knowledge-others/install-unity-version/

**Reply by dam-ease** [reply]
Hello @asoteamgen,
Any update for me on this? If there is anything else I can assist you with, please don't hesitate to let me know.
